jetty-servlets/src/main/java/org/eclipse/jetty/servlets/IncludeExcludeBasedFilter.java

```diff
@@ -110,8 +114,70 @@ public void init(FilterConfig filterConfig) throws ServletException
         }
     }
 
+    @Override
+    public final void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException
+    {
+        HttpServletRequest http_request = (HttpServletRequest)request;
+        HttpServletResponse http_response = (HttpServletResponse)response;
+        // the mime type may not be available before the filter chain is invoked, so applying the filter may result in incorrect mime type logic if done
+        // before chain.doFilter
+        // the response may have been already committed after the filter chain is invoked, so applying the filter may not do anything if done after
+        // chain.doFilter
+        // thus we wrap the response to apply the filter immediately before it is committed
+        chain.doFilter(request,new HttpServletResponseWrapper(http_response)
```

Comment:
No.  Don't create a response wrapper.
That will break too many optimizations in the rest of Jetty.